### PR TITLE
Allow const_dt with EM and check it for ES

### PR DESCRIPTION
**Pull request by dpgrote**
This small PR allows more control of the time step for EM simulations and ensures that it is set for ES.

### CODE REVIEW

**Comment by ax3l** [comment]
@dpgrote it looks like `embedded_boundary_cube` and `embedded_boundary_cube` crash with the current PR diff.

File: Source/WarpX.H [hunk 1]
```diff
@@ -1339,7 +1339,8 @@ private:
     int num_injected_species = -1;
     amrex::Vector<int> injected_plasma_species;
 
-    amrex::Real const_dt = amrex::Real(0.5e-11);
+    amrex::Real const_dt;
+    bool const_dt_specified;
```
Comment (ax3l) [inline Source/WarpX.H]: Let's use a single `std::optional<amrex::Real>` as type and call the member variable `m_const_dt` according to our conventions?

Comment (ax3l) [reply Source/WarpX.H]: ping @dpgrote when you have the time :)

**Comment by RemiLehe** [comment]
I just updated the PR according to @ax3l's comments.

**Review by RemiLehe — APPROVED** [review]

**Pull request merged**